src/routes/blog/post/introducing-python-machine-learning-runtime/+page.markdoc

```diff
@@ -0,0 +1,117 @@
+
+## Getting started
+
+1. Head to the [Appwrite Console](https://cloud.appwrite.io/console) then click on **Functions** in the left sidebar and then click on the **Create Function** button.
+    ![Create Function](/images/blog/introducing-python-machine-learning-runtime/create-function.png)
+2. To get started with the new runtime, navigate to your Appwrite console and select `Python (ML) 3.11` when creating a new function. 
+3. Follow the step-by-step wizard and create the function.
+
+Here's an example function that trains a simple neural network on the MNIST dataset using TensorFlow and TensorFlow Datasets:
+
+```python
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+
+def normalize_img(image, label):
+    return tf.cast(image, tf.float32) / 255.0, label
+
+
+def main(context):
+    (ds_train, ds_test), ds_info = tfds.load(
+        "mnist",
+        split=["train", "test"],
+        shuffle_files=True,
+        as_supervised=True,
+        with_info=True,
+    )
+
+    ds_train = ds_train.map(normalize_img, num_parallel_calls=tf.data.AUTOTUNE)
+    ds_train = ds_train.cache()
+    ds_train = ds_train.shuffle(ds_info.splits["train"].num_examples)
+    ds_train = ds_train.batch(128)
+    ds_train = ds_train.prefetch(tf.data.AUTOTUNE)
+
+    ds_test = ds_test.map(normalize_img, num_parallel_calls=tf.data.AUTOTUNE)
+    ds_test = ds_test.batch(128)
+    ds_test = ds_test.cache()
+    ds_test = ds_test.prefetch(tf.data.AUTOTUNE)
+
+    model = tf.keras.models.Sequential(
+        [
+            tf.keras.layers.Flatten(input_shape=(28, 28)),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dense(10),
+        ]
+    )
+
+    model.compile(
+        optimizer=tf.keras.optimizers.Adam(0.001),
+        loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+        metrics=[tf.keras.metrics.SparseCategoricalAccuracy()],
+    )
+
+    model.fit(
+        ds_train,
+        epochs=6,
+        validation_data=ds_test,
+    )
+
+    return context.res.json({"success": True})
+```
+
+Find the full source code in our [GitHub repository](https://github.com/open-runtimes/open-runtimes/tree/main/runtimes/python-ml-3.11).
+
+## Best practices
+
+The Python ML runtime is in *beta.* We plan to iterate in response to developer demand and needs. For now, the runtime is that it is CPU-only. Here are some tips and tricks to help you overcome this limitation:
+
+### Smaller models & Tensorflow Lite
+
+Smaller models are faster to load and execute. Avoid using large models that require a lot of memory and CPU resources.
+
+Tensorflow Lite models are optimized for mobile and edge devices. They are smaller in size and can be run on CPUs. You can convert your Tensorflow models to Tensorflow Lite models using the [Tensorflow Lite Converter](https://www.tensorflow.org/lite/convert).
```

Comment:
```suggestion
Smaller models are faster to load and execute. Avoid using large models that require a lot of memory and compute resources.

Tensorflow Lite models are optimized for mobile and edge devices. They are smaller in size and can be run on CPUs. You can convert your Tensorflow models to Tensorflow Lite models using the [Tensorflow Lite Converter](https://www.tensorflow.org/lite/convert).
```

Maybe we can expand on the point of this conversion and share what exactly happens during the conversion to tensorflow lite. Some before vs after size & resource usage comparisons from the tensorflow docs etc to basically drive home the point. Tensorflow Lite models are almost as good as the real deal, but with some variations in accuracy

Reply:
Couldn't find anything about tensorflow vs lite comparisons on their docs, but added some more explanation